final/problem-def-ana/problem-def.tex

```diff
@@ -1,29 +1,38 @@
 \section{Problem Definition}
 \label{sec:problem-definition}
 
-The previous sections provided the necessary background knowledge of
-the problem domain. Now that the background knowledge has been
-explained, this section will discuss the problem definition of the
-client.
+Now that the background knowledge has been explained, this section discusses the
+problem definition of the client.
 
-Spoofax offers a wide variety of tools to develop DSLs and
-accompanying IDE support. Precisely because Spoofax is concerned with
-developing DSLs, rapid prototyping of syntax and grammar is a
-convenient addition to the product. \Cref{sec:repl} showed that REPLs
-provide this rapid prototyping ability for other languages like Lisp
-and Python. Therefore, the client has expressed their interest in
-a REPL that works with all language definitions created in
-Spoofax. Such a REPL would aid both the developer of the language and
-its end-user.
+Spoofax offers a wide variety of tools to develop DSLs and accompanying IDE
+support. Precisely because Spoofax is concerned with developing DSLs, rapid
+prototyping of syntax and grammar is a convenient addition to the product.
+\Cref{sec:repl} showed that REPLs provide precisely this rapid prototyping
+ability.  Therefore, the client has expressed their interest in a REPL that
+works with languages designed with Spoofax. Such a REPL would aid both the
+developer of the language and its end-user. The main problem, thus, can be
+stated as a question as follows: \textit{``How can a REPL be created that can be
+used with languages defined with Spoofax?''}.
 
-Since a lot of language services are already provided by Spoofax, the
-REPL should try to hook into existing services as much as possible.
-Besides exposing already existing Spoofax services in a REPL,
-the product should also expose additional features geared
-towards interactive exploration of programs in Spoofax defined
-languages.  Examples include displaying and editing the contents of
-current program context, saving and loading program context and
-keeping a history of executed expressions.
+Spoofax already provides a lot of language services that can be reused. However,
+this is not without issues, as Spoofax's infrastructure currently is not geared
+towards interactive use. A subquestion, then, is: \textit{``How can Spoofax's
+services be changed to support both interactive and non-interactive use?''}.
+
+% TODO: Java-backend example requires a cite?
+Additionally, there are currently two ways to define the dynamic semantics of a
+language (see \cref{ssec:dynamic-semantics}); one can either use DynSem or Stratego.
+Some languages even define their own interpreter writtin in Java. Therefore,
```

Comment:
I think this should be phrased differently: a language does not define anything. Maybe:

> Some languages even have an interpreter implemented in Java